onadata/libs/utils/export_builder.py

```diff
@@ -77,10 +77,8 @@ def get_value_or_attachment_uri(
             if a.get('filename') and a.get('filename').endswith(value)
         ]
         if attachments:
-            value = current_site_url(
-                attachments[0].get('download_url', '')
-            )
-
+            value = current_site_url(attachments[0].get('download_url', ''))
```

Comment:
I don't think this addresses the issue, as we had discussed the check needs to be applied on line 77, replacing it with:
```python
if a.get('name') == value
```
The current issue is that `filename` already has the spaces as underscores. We should be using the newly introduced `name` column that will always have the exact same name as the `value` when it was captured during data collection.